test/pkg/environment/aws/default_nodepool.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+ name: default
+spec:
+ disruption:
+   consolidationPolicy: WhenEmptyOrUnderutilized
+   consolidateAfter: Never
+   budgets:
+     - nodes: 100%
+ limits:
+   cpu: 1000
+   memory: 1000Gi
+ template:
+   spec:
+     expireAfter: Never
+     requirements:
+       - key: kubernetes.io/os
+         operator: In
+         values: ["linux"]
+       - key: karpenter.sh/capacity-type
+         operator: In
+         values: ["on-demand"]
+       - key: karpenter.k8s.aws/instance-category
+         operator: In
+         values: ["c", "m", "r"]
+       - key: karpenter.k8s.aws/instance-generation
+         operator: In
+         values: ["2"]
+       - key: karpenter.k8s.aws/instance-family
+         operator: NotIn
+         values: ["a1"]
+     nodeClassRef:
+      group: eks.amazonaws.com
```

Comment:
Should this be referencing an EKS auto mode nodeclass? I imagine it should be referencing a karpenter.k8s.aws/EC2NodeClass.